hudi-sync/hudi-sync-common/src/main/java/org/apache/hudi/sync/common/AbstractSyncHoodieClient.java

```diff
@@ -136,6 +141,42 @@ public MessageType getDataSchema() {
     }
   }
 
+  public abstract static class TypeOptimizer implements Serializable {
```

Comment:
TypeConverter looks better? I see the function is to convert type and would you please add some tests?

Reply:
@leesf Of course. I will add tests  for it ASAP.

Reply:
+1

Reply:
@vinothchandar @leesf I added some tests based on review comments. Please take a look.
Thank you.